src/main/java/org/gradle/profiler/buildops/BuildOperationInstrumentation.java

```diff
@@ -1,31 +1,71 @@
 package org.gradle.profiler.buildops;
 
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
 import org.gradle.profiler.instrument.GradleInstrumentation;
 
-import java.io.*;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
 import java.time.Duration;
+import java.util.Map;
 import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
 
 public class BuildOperationInstrumentation extends GradleInstrumentation {
-    private final File dataFile;
+    private final File configurationTimeDataFile;
+    private final ImmutableMap<String, File> buildOperationDataFiles;
+    private final boolean measureConfigTime;
 
-    public BuildOperationInstrumentation() throws IOException {
-        dataFile = File.createTempFile("gradle-profiler", "build-ops");
-        dataFile.deleteOnExit();
+    public BuildOperationInstrumentation(boolean measureConfigTime, ImmutableList<String> measuredBuildOperations) throws IOException {
+        this.measureConfigTime = measureConfigTime;
+        this.configurationTimeDataFile = File.createTempFile("gradle-profiler", "build-ops-config-time");
+        this.configurationTimeDataFile.deleteOnExit();
+        this.buildOperationDataFiles = measuredBuildOperations.stream()
+            .collect(ImmutableMap.toImmutableMap(Function.identity(), BuildOperationInstrumentation::createOpTmpFile));
```

Comment:
Does it make sense to put each operation type into its own file? What is the performance impact of this?